### PR TITLE
Multiple small changes

### PR DESCRIPTION
**Pull request by kduske-n4**
Sorry to submit multiple changes in one pull request (and not using branches), but I had to make these changes in quick succession and could not isolate them. They are independent of each other, however. There are three changes:

**Convert nested Java maps to Ruby hashes in `RubyHashUtil#toRubyObject`**
This prevents a `ClassCastException` when, after creating a new block in a block processor, you attempt to access the attributes of that block later in a tree processor. When asking the block for its attributes, AsciidoctorJ assumes that what it gets from the asciidoctor AST is an instance of `RubyHash`, when in this case what it really gets is an instance of ` MapJavaProxy` that was injected by JRuby into the asciidoctor AST when the block was created originally.

I'm not sure about this change, since I had to choose one of several different methods to convert Java maps to Ruby hashes in `RubyHashUtil`.

**Return `null` if the inner document of a table cell is `nil`**
This prevents a `NullPointerException` when attempting to access the inner document of a table cell in a processor when that cell doesn't have an inner document in the asciidoctor AST.

**Add getter for `@content_model`**
This adds a getter for the `@content_model` attribute of an asciidoctor `AbstractBlock` to `StructuralNode`.

I hope these changes are useful for you and that you can apply them. I have run the tests in the asciidoctorj-asciidoctorj and asciidoctorj-distribution projects and there were no failures.

### CODE REVIEW

File: asciidoctorj-core/src/main/java/org/asciidoctor/ast/impl/CellImpl.java [hunk 1]
```diff
@@ -70,7 +70,10 @@ public void setVerticalAlignment(Table.VerticalAlignment valign) {
 
     @Override
     public Document getInnerDocument() {
-        return (Document) NodeConverter.createASTNode(getRubyProperty("inner_document"));
+        IRubyObject innerDocument = getRubyProperty("inner_document");
+        if (innerDocument.isNil())
+        	return null;
+		return (Document) NodeConverter.createASTNode(innerDocument);
```
Comment (robertpanzer) [inline asciidoctorj-core/src/main/java/org/asciidoctor/ast/impl/CellImpl.java]: I haven't tested it, but I don't think this will compile.

**Comment by robertpanzer** [comment]
Thanks a lot for your great contribution, Kristian!

Would you mind adding tests for the three problems you fixed?
That should be 
- a test that simply invokes getContentModel (Maybe like here: https://github.com/asciidoctor/asciidoctorj/blob/07fed407fce84dc9c0f7cf203eb8266255ebad7b/asciidoctorj-core/src/test/java/org/asciidoctor/WhenAsciiDocIsRenderedToDocument.java#L149)
- a test that uses two extensions, where one creates a node and another that tries to access the attributes? (Maybe copy this test and accommodate to the needs: https://github.com/asciidoctor/asciidoctorj/blob/07fed407fce84dc9c0f7cf203eb8266255ebad7b/asciidoctorj-core/src/test/groovy/org/asciidoctor/extension/WhenABlockMacroProcessorCreatesATable.groovy#L51)
- A test that tries to access the inner document of a table cell when it doesn't have the Asciidoctor style. That should fit in here: https://github.com/asciidoctor/asciidoctorj/blob/asciidoctorj-1.6.0/asciidoctorj-core/src/test/groovy/org/asciidoctor/WhenATableIsLoaded.groovy

That would help us a lot and would avoid that we later stumble over these problems again.

File: asciidoctorj-core/src/main/java/org/asciidoctor/ast/impl/CellImpl.java (hunk 1 above)
Comment (kduske-n4) [inline asciidoctorj-core/src/main/java/org/asciidoctor/ast/impl/CellImpl.java]: It does, though. Why shouldn't it?

**Comment by kduske-n4** [comment]
Added test cases as requested.

File: asciidoctorj-core/src/main/java/org/asciidoctor/ast/impl/CellImpl.java (hunk 1 above)
Comment (robertpanzer) [inline asciidoctorj-core/src/main/java/org/asciidoctor/ast/impl/CellImpl.java]: Gotcha! :-)
Could you please add braces around the `return null`.

I probably automatically put them in there mentally around both returns that have the same indentation.

Comment (kduske-n4) [inline asciidoctorj-core/src/main/java/org/asciidoctor/ast/impl/CellImpl.java]: Done.

**Comment by robertpanzer** [comment]
Thanks a lot! Such tests really help us.
Executed them locally and they look good, if I revert the main code, I can also see the errors that you fixed.

There are a couple of code style problems that codenarc complains.
Would you mind fixing them?
While it might seem pedantic it helps us to keep a common code style across the groovy code.
When doing a `./gradlew clean build` codenarc will create a report that shows you what it doesn't like.

Thanks a lot!

Cheers
Robert

**Comment by kduske-n4** [comment]
Sure, will do!

**Comment by kduske-n4** [comment]
Done ;-)

**Comment by kduske-n4** [comment]
Some things still seem broken in your CI. I have run the tests locally and it was fine. Any hints?

**Comment by robertpanzer** [comment]
Yes, there are some problems with Asciidoctor PDF and JRuby on Windows.
That's why this test is in. :-)
Will upgrade the JRuby version soon and look if it works better now (even though I don't have much hope)
Unix build is green, so I'm fine to merge.

Thanks!

**Pull request merged**